File: lodestar-api/src/test/java/com/oayilix/lodestar/mapping/LodestarMapping.java
package com.oayilix.lodestar.mapping;

import android.app.Activity;
import java.util.Collections;
import java.util.LinkedHashMap;
import java.util.Map;

public final class LodestarMapping {
    private LodestarMapping() {}

    public static Map<String, Class<? extends Activity>> get() {
        Map<String, Class<? extends Activity>> routes = new LinkedHashMap<>();
        routes.put("lodestar://example.com/app/test", TestActivity.class);
        return Collections.unmodifiableMap(routes);
    }

    public static class TestActivity extends Activity {}
}

File: lodestar-api/src/test/kotlin/com/oayilix/lodestar/mapping/LodestarMapping.kt
package com.oayilix.lodestar.mapping

import android.app.Activity

/**
 * Test-only aggregate mapping that mirrors the Gradle plugin generated entry point.
 * 仅测试使用的聚合映射表，用于模拟 Gradle 插件生成的入口类。
 */
object LodestarMapping {

    /**
     * Exposes the same static `get()` method shape as the production generated class.
     * 暴露与生产环境生成类一致的静态 `get()` 方法形态。
     */
    @JvmStatic
    fun get(): Map<String, Class<out Activity>> =
        mapOf("lodestar://example.com/app/test" to TestActivity::class.java)

    class TestActivity : Activity()
}
